machine/machine_libipmctl.go

```diff
@@ -0,0 +1,69 @@
+// +build libipmctl
+
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package machine
+
+// #cgo pkg-config: libipmctl
+// #include <nvm_management.h>
+import "C"
+import (
+	"fmt"
+
+	"k8s.io/klog"
+)
+
+// GetNVMAvgPowerBudget retrieves configured power budget
+// (in watts)for NVM devices. When libipmct is not available
+// zero is returned.
+func GetNVMAvgPowerBudget() (uint, error) {
+	// Initialize libipmctl library.
+	err := C.nvm_init()
+	if err != C.NVM_SUCCESS {
+		klog.Warningf("libipmctl initialization failed with status %d", err)
+		return 0, fmt.Errorf("libipmctl initialization failed with status %d", err)
```

Comment:
in other places in this function you return uint(0)

Reply:
I have *no* idea why it compiles, to be honest O.o